### ISSUE TITLE
Crash on clearing breadcrumbs

### ISSUE THREAD

**Issue by luisguerrerolab** [MindscapeHQ/raygun4xamarin-samples#9]
### Summary

On some scenarios, the package causes a crash after calling `RaygunClient.Current.ClearBreadcrumbs();` that is reported in the Raygun dashboard with the message:

`index + length > size`

The project is running `Raygun4Xamarin.Forms` v1.0.3.

Reports show that this crash is generated in both platforms, Android & iOS.

### Relevant Logs
```
...
     "Error": {
          "Data": {},
          "ClassName": "System.IndexOutOfRangeException",
          "Message": "index + length > size",
          "StackTrace": [
               {
                    "LineNumber": 0,
                    "ClassName": "System.Array.Clear (System.Array array, System.Int32 index, System.Int32 length) <0x1056a0d90 + 0x000e4>",
                    "FileName": "<aedd57cfa7174b1b89f01dd6015f4050#6514dc30122f6a91822dd1e2431ece39>",
                    "Raw": "at System.Array.Clear (System.Array array, System.Int32 index, System.Int32 length) <0x1056a0d90 + 0x000e4> in <aedd57cfa7174b1b89f01dd6015f4050#6514dc30122f6a91822dd1e2431ece39>:0"
               },
               {
                    "LineNumber": 0,
                    "ClassName": "System.Collections.Generic.Queue`1[T].Clear () <0x105831100 + 0x00053>",
                    "FileName": "<aedd57cfa7174b1b89f01dd6015f4050#6514dc30122f6a91822dd1e2431ece39>",
                    "Raw": "at System.Collections.Generic.Queue`1[T].Clear () <0x105831100 + 0x00053> in <aedd57cfa7174b1b89f01dd6015f4050#6514dc30122f6a91822dd1e2431ece39>:0"
               },
               {
                    "LineNumber": 0,
                    "ClassName": "Raygun4Xamarin.Forms.RaygunCrashReporter.ClearBreadcrumbs () <0x1071db0e0 + 0x0001f>",
                    "FileName": "<b3db72b1f0c14730add77a9461e19246#6514dc30122f6a91822dd1e2431ece39>",
                    "Raw": "at Raygun4Xamarin.Forms.RaygunCrashReporter.ClearBreadcrumbs () <0x1071db0e0 + 0x0001f> in <b3db72b1f0c14730add77a9461e19246#6514dc30122f6a91822dd1e2431ece39>:0"
               },
               {
                    "LineNumber": 0,
                    "ClassName": "Raygun4Xamarin.Forms.RaygunClient.ClearBreadcrumbs () <0x1071dfa60 + 0x00037>",
                    "FileName": "<b3db72b1f0c14730add77a9461e19246#6514dc30122f6a91822dd1e2431ece39>",
                    "Raw": "at Raygun4Xamarin.Forms.RaygunClient.ClearBreadcrumbs () <0x1071dfa60 + 0x00037> in <b3db72b1f0c14730add77a9461e19246#6514dc30122f6a91822dd1e2431ece39>:0"
               },
...
```

I'd like to share more details, but unfortunately I haven't been able to reproduce this issue. However, we've found 7 crash reports so far and they happened under different circumstances.

**Comment by mduncan26** [comment]
Hi @albertoms 

Thank you for getting in touch with us. I will make a ticket for this issue for us to consider in future work. My hunch is this may be due to the Raygun client not being as async safe as it should be. 

We will be back in contact once this work has been completed.

Thank you,
Mitchell.

**Comment by mduncan26** [comment]
Hi @luisguerrerolab 

Just following up to let you know we released a fix for this issue in v1.0.6 of the Raygun4Xamarin.Forms package. Please let me know if you experience anymore issue.

https://www.nuget.org/packages/Raygun4Xamarin.Forms/1.0.6

Thank you,
Mitchell.
